code/modules/power/generator.dm

```diff
@@ -9,43 +9,19 @@
 	var/obj/machinery/atmospherics/components/binary/circulator/cold_circ
 	var/obj/machinery/atmospherics/components/binary/circulator/hot_circ
 
-	//note: these currently only support EAST and WEST
-	var/cold_dir = WEST
-	var/hot_dir = EAST
-
 	var/lastgen = 0
 	var/lastgenlev = -1
 	var/lastcirc = "00"
 
 
 /obj/machinery/power/generator/Initialize(mapload)
 	. = ..()
-	var/obj/machinery/atmospherics/components/binary/circulator/circpath = /obj/machinery/atmospherics/components/binary/circulator
-	cold_circ = locate(circpath) in get_step(src, cold_dir)
-	hot_circ = locate(circpath) in get_step(src, hot_dir)
+	find_circs()
 	connect_to_network()
 	SSair.atmos_machinery += src
-
-	if(cold_circ)
-		switch(cold_dir)
-			if(EAST)
-				cold_circ.side = circpath.CIRC_RIGHT
-			if(WEST)
-				cold_circ.side = circpath.CIRC_LEFT
-		cold_circ.update_icon()
-
-	if(hot_circ)
-		switch(hot_dir)
-			if(EAST)
-				hot_circ.side = circpath.CIRC_RIGHT
-			if(WEST)
-				hot_circ.side = circpath.CIRC_LEFT
-		hot_circ.update_icon()
-
-	if(!cold_circ || !hot_circ)
-		stat |= BROKEN
-
 	update_icon()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/machine/generator(null)

```

Comment:
Same as above

Reply:
again, deep fryer does it this way but if you want me to change it I will

Reply:
the deep fryer is not a model for anything, do as cyberboss suggests